src/datasource.ts

```diff
@@ -807,28 +807,34 @@ export default class IrondbDatasource extends DataSourceApi<IrondbQueryInterface
 
     throwerr(error: any) {
         log(() => 'throwerr() err = ' + JSON.stringify(error));
+
         if (typeof error !== 'string') {
-            if (error.data) {
-                if (error.data.message) {
-                    var message;
-                    try {
-                        message = JSON.parse(error.data.message);
-                    } catch (error) {
-                        log(() => 'throwerr() failed to parse json from error.data.message. error: ' + error);
-                    }
-                    if (message.user_error) {
+            if (error.data && error.data.message) {
+                var message
+                try {
+                    message = JSON.parse(error.data.message);
+                } catch {
+                    throw String(error.status + ': ' + error.statusText)
+                }
+                if (message.user_error) {
+                    if (message.arguments && message.arguments.q) {
                         throw message.user_error.message + ', in query: ' + message.arguments.q;
-                    } else if (error.statusText === 'Not Found') {
-                        throw 'Circonus IRONdb Error: ' + error.statusText;
-                    } else if (error.statusText && error.status > 0) {
-                        throw 'Network Error: ' + error.statusText + '(' + error.status + ')';
+                    } else {
+                        throw message.user_error
                     }
                 }
-            } else {
-                throw String(error);
+                else {
+                    throw JSON.stringify(message)
+                }
             }
-        } else {
-            throw error ? error.toString() : 'unknown';
+            else {
+                throw JSON.stringify(error)
+            }
+        } else if (typeof error === 'string'){
+            throw JSON.stringify(error)
+        }
+        else {
+            throw JSON.stringify(error)
         }
```

Comment:
This is missing a few things.   I think this is what you are looking for (please check it though).
```
        if (typeof error === 'object') {
            if (error.data && error.data.message) {
                var message
                try {
                    message = JSON.parse(error.data.message);
                } catch {
                    throw String(error.status + ': ' + error.statusText)
                }
                if (typeof message === 'object' && message.user_error) {
                    if (message.arguments && message.arguments.q && message.user_error.message) {
                        throw message.user_error.message + ', in query: ' + message.arguments.q;
                    } else {
                        throw message.user_error
                    }
                }
                else {
                    throw JSON.stringify(message)
                }
            }
            else {
                throw JSON.stringify(error)
            }
        } 
        else {
            throw JSON.stringify(error)
        }
```